src/main/java/io/cdap/plugin/salesforce/SalesforceSchemaUtil.java

```diff
@@ -147,20 +148,49 @@ public static void checkCompatibility(Schema actualSchema, Schema providedSchema
     }
   }
 
-  public static Schema getSchemaWithFields(SObjectDescriptor sObjectDescriptor, SObjectsDescribeResult describeResult) {
+  public static Schema getSchemaWithFields(SObjectDescriptor sObjectDescriptor,
+                                           SObjectsDescribeResult describeResult) {
+    return getSchemaWithFields(sObjectDescriptor, describeResult, Collections.emptyList());
+  }
+
+  public static Schema getSchemaWithFields(SObjectDescriptor sObjectDescriptor,
+                                           SObjectsDescribeResult describeResult,
+                                           List<String> topLevelParents) {
     List<Schema.Field> schemaFields = new ArrayList<>();
 
     for (SObjectDescriptor.FieldDescriptor fieldDescriptor : sObjectDescriptor.getFields()) {
-      String parent = fieldDescriptor.hasParents() ? fieldDescriptor.getLastParent() : sObjectDescriptor.getName();
-      Field field = describeResult.getField(parent, fieldDescriptor.getName());
-      if (field == null) {
-        throw new IllegalArgumentException(
-          String.format("Field '%s' is absent in Salesforce describe result", fieldDescriptor.getFullName()));
+
+      SalesforceFunctionType functionType = fieldDescriptor.getFunctionType();
+      Schema fieldSchema = null;
+      if (!functionType.isConstant()) {
+        List<String> allTopLevelParents = new ArrayList<>(topLevelParents);
+        allTopLevelParents.add(sObjectDescriptor.getName());
+        String parentsPath = fieldDescriptor.getParentsPath(allTopLevelParents);
+        Field field = describeResult.getField(parentsPath, fieldDescriptor.getName());
+
+        if (field == null) {
+          throw new IllegalArgumentException(
+            String.format("Field '%s' is absent in Salesforce describe result", fieldDescriptor.getFullName()));
```

Comment:
include the name of the sobject in the error message, something like:

Field '%s' does not exist in Salesforce object '%s'.